### ISSUE TITLE
Parsing loci from exome file is slow

### ISSUE THREAD

**Issue by arahuja** [hammerlab/guacamole#285]
I am trying to pass a file with exome capture positions as the `--loci-from-file` argument, but currently it just hangs.

From some profiling, most of this time seems to be spent in joining the individual `LociSet` created from each capture region. 

```
Common.scala:198 org.hammerlab.guacamole.LociSet$.parse(String) 149869
LociSet.scala:148 org.hammerlab.guacamole.LociSet$.union(Seq) 146856
LociSet.scala:153 org.hammerlab.guacamole.LociSet.union(LociSet) 146768
LociSet.scala:53 org.hammerlab.guacamole.LociMap.union(LociMap) 146752
LociMap.scala:81 org.hammerlab.guacamole.LociMap$.union(Seq) 146727
LociMap.scala:255 org.hammerlab.guacamole.LociMap$Builder.result() 139540
LociMap.scala:214 org.hammerlab.guacamole.LociMap$Builder$$anonfun$result$1$$anonfun$apply$4.apply(Tuple3) 86963
LociMap.scala:229 com.google.common.collect.TreeRangeMap.put(Range, Object) 61620
```

@timodonnell do you think this is worth optimizing or just handling this differently?  Does anything pop-out immediately as a way to fix this otherwise I can take it on.  I think we had talked about this earlier on- that it would be difficult to handle many small regions this way?

**Comment by timodonnell** [comment]
I think `LociSet.parse` could probably be implemented more efficiently. Can you post the intervals file you're using somewhere?

**Comment by arahuja** [comment]
I posted the file https://gist.github.com/arahuja/54fa61d073ed2a44ee7d

Also, I was just thinking supporting VCF or BED files directly would be useful as well.

**Comment by timodonnell** [comment]
Thanks, I'll take a quick crack at it now and see if I can speed it up.

Agree that VCF / BED could be a useful feature.

**Comment by timodonnell** [comment]
Attempted fix in #286

**Comment by arahuja** [comment]
Great, this worked thanks!

Closed by #286
